src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtil.java

```diff
@@ -62,19 +62,18 @@ public static void generate(String... args) throws IOException, CheckstyleExcept
         defaultConfiguration.addChild(treeWalkerConfig);
         treeWalkerConfig.addChild(scraperCheckConfig);
         checker.configure(defaultConfiguration);
-        dumpMetadata(checker, args[0]);
+        final List<File> validFiles = extractValidFiles(args[0]);
+        checker.process(validFiles);
     }
 
     /**
-     * Process files using the checker passed and write to corresponding XML files.
+     * Extracts valid files using the path passed.
      *
-     * @param checker checker
      * @param path rootPath
-     * @throws CheckstyleException checkstyleException
+     * @return a list of valid files
      * @throws IOException ioException
      */
-    private static void dumpMetadata(Checker checker, String path) throws CheckstyleException,
-            IOException {
+    public static List<File> extractValidFiles(String path) throws IOException {
```

Comment:
We should funnel the users to 1 public method. It will start be confusing if we have multiple and are not clear on which does what. For example, why would I call `extract` when all I want to do is `generate` the files, and then this leads back to why isn't `generate` reporting errors to me.